widgets/pkg_all.lua

```diff
@@ -33,16 +33,34 @@ local function worker(format, warg)
     }
 
     -- Check if updates are available
+    local function parse(str, skiprows)
+        local size, lines, first = 0, "", skiprows or 0
+        for line in str:gmatch("[^\r\n]+") do
+            if size >= first then
+                lines = lines .. (size == first and "" or "\n") .. line
+            end
+            size = size + 1
+        end
+        size = math.max(size-first, 0)
+        return {size, lines}
+    end
+    
+    -- Select command
     local _pkg = manager[warg]
-    local f = io.popen(_pkg.cmd)
+    spawn.easy_async(_pkg.cmd, function(stdout) callback(parse(stdout, _pkg.sub)) end)
+end
+-- }}}
 
-    for line in f:lines() do
-        updates = updates + 1
-    end
-    f:close()
+-- {{{ Packages widget type
+local function worker(format, warg)
+    local ret = nil
+    
+    pkg_all.async(warg, function(data) ret = data end)
 
-    return {_pkg.sub and math.max(updates-_pkg.sub, 0) or updates}
+    while ret==nil do end
```

Comment:
Would it be better to give the `async` function an callback function from `init.lua`? So it could update the widget itself and we would not need to waste CPU cycles ... (don't know how handy and implementable this is)